nav2_behavior_tree/include/nav2_behavior_tree/plugins/speed_controller.hpp

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2018 Intel Corporation
+// Copyright (c) 2020 Sarthak Mittal
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__PLUGINS__SPEED_CONTROLLER_HPP_
+#define NAV2_BEHAVIOR_TREE__PLUGINS__SPEED_CONTROLLER_HPP_
+
+#include <memory>
+#include <string>
+#include <deque>
+
+#include "nav_msgs/msg/odometry.hpp"
+#include "nav2_util/odometry_utils.hpp"
+
+#include "behaviortree_cpp_v3/decorator_node.h"
+
+namespace nav2_behavior_tree
+{
+
+class SpeedController : public BT::DecoratorNode
+{
+public:
+  SpeedController(
+    const std::string & name,
+    const BT::NodeConfiguration & conf);
+
+  // Any BT node that accepts parameters must provide a requiredNodeParameters method
+  static BT::PortsList providedPorts()
+  {
+    return {
+      BT::InputPort<double>("min_rate", 0.1, "Minimum rate"),
```

Comment:
Because this has been sitting around for so long, we need to add this to our new docs. https://github.com/ros-planning/navigation.ros.org/pull/25 and https://github.com/ros-planning/navigation2/blob/master/doc/parameters/param_list.md